asv/console.py

```diff
@@ -304,6 +304,13 @@ def enable(self, verbose=False):
         else:
             self._logger.setLevel(logging.INFO)
 
+    @contextlib.contextmanager
+    def set_level(self, level):
+        orig_level = self._logger.level
+        self._logger.setLevel(level)
+        yield
+        self._logger.setLevel(orig_level)
```

Comment:
I've been using this PR for the Scipy benchmarks, and didn't meet other
problems than what is fixed above.